source/pod.go

```diff
@@ -104,43 +99,46 @@ func (ps *podSource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, error
 			}
 		}
 
-		if domainAnnotation, ok := pod.Annotations[hostnameAnnotationKey]; ok {
-			domainList := splitHostnameAnnotation(domainAnnotation)
-			for _, domain := range domainList {
-				if len(targets) == 0 {
-					node, _ := ps.nodeInformer.Lister().Get(pod.Spec.NodeName)
-					for _, address := range node.Status.Addresses {
-						recordType := suitableType(address.Address)
-						// IPv6 addresses are labeled as NodeInternalIP despite being usable externally as well.
-						if address.Type == corev1.NodeExternalIP || (address.Type == corev1.NodeInternalIP && recordType == endpoint.RecordTypeAAAA) {
-							addToEndpointMap(endpointMap, domain, recordType, address.Address)
+		// accept annotations that point to the node's external IP (or IPv6 NodeInternalIP) only for pods using the host network
+		if pod.Spec.HostNetwork {
```

Comment:
Wdyt about removing this condition to cover all cases described in https://github.com/kubernetes-sigs/external-dns/pull/3468#issuecomment-1537556184 ?

Reply:
I can include that change in here, will do so later today.

Reply:
@mloiseleur please check, I added the changes and also some more docs updates.

Reply:
It's lgtm at first glance. I asked @gregorycuellar to also check, if possible.